### ISSUE TITLE
No Man's Connect is protecting my username from me

### ISSUE THREAD

**Issue by taukausanake** [jaszhix/NoMansConnect#35]
I just updated NMC to the latest build (v0.14.0) and the program has a window pop up saying "Protection Enabled For undefined". After that the program is stuck on the circle spinning loading screen unless I click "Sync Locations" after which the normal views come up. I did enable username protection when I first installed but I was under the assumption that it would save it for my computer even after an update. Username protection option is also missing now. Trying to override username to my normal one still comes up as protected from me. 
I am running Windows 10 x64. My NMS install is in my Steam library, normal path.

On the side: How does one teleport using the program? I can't find a command or anything in the program. Do I need NMS running in order to teleport and if so how does it work? Could you put instructions somewhere please?

**Comment by jaszhix** [comment]
I need your username. There are instructions when you click a menu on the upper left corner of the location box (...) ellipsis and hovering over "Teleport" tells you what to do (click it). You also seemed to have stumbled on to a bug with the dialog for undefined user.

**Comment by taukausanake** [comment]
My username is Tauka Usanake. Are there any files you want me to copy and send over to you?

I did not see any option to teleport when I clicked the ellipse, at least not when I selected a spot on the map and click the location box that comes up on the bottom left. I will double check today though

**Comment by jaszhix** [comment]
Seeing your settings.json and NMC.log couldn't hurt, but if its a hassle no worries. I got it unprotected for you.

**Comment by jaszhix** [comment]
I just ran a Windows 10 upgrade and my machine ID is the same. If you're willing to provide more info on this it might be helpful.

**Comment by taukausanake** [comment]
NMC.log - expires in one day
https://pastebin.com/eawzG8Qw

settings.json - also expires in a day
https://pastebin.com/7GQ4fmrh

It wasn't my machine getting upgraded but the NMC program

**Comment by jaszhix** [comment]
Thanks. Which version did you upgrade from? If its pre-0.12, then this is a settings bug that probably won't occur again (I migrated settings to JSON). 

Side note - the log shouldn't have any sensitive data, the machine ID is actually a hashed version of the real ID.

**Comment by taukausanake** [comment]
I previously had 0.13.0 and upgraded to 0.14.0, the current build. I am still getting the error at startup.
![image](https://user-images.githubusercontent.com/1978025/28149424-1dc2c50c-6753-11e7-803d-6eab10f74e08.png)
After which a Sync Locations brings the map and explored locations up. I do not see a Teleport option on the location box still, with or without the game running. The app does know where my NMS install is and has mapped my starting location prior to updating to 0.14.0
![image](https://user-images.githubusercontent.com/1978025/28149512-bb0a0276-6753-11e7-8150-ffb65fc650e4.png)

**Comment by ishKiia** [comment]
Same thing happening to me .. username it is screaming about is "ishKiia"

**Comment by taukausanake** [comment]
I have an update, issue might be my fault one way or another because of the PS4 support thing before. I have my platform set to PS4 because that's the main version I want to play. I have a PC install (thank you Waking Titan) but only intend to use it for exploring of the systems from my PS4 save, hence my desire to use this tool.
So after I set platform from PS4 back to PC I don't get the issue anymore. I switch it back to PS4 and the issue is there. I take it you use the same username pool for both databases.
I take it this is also why I can't use teleport while my platform is on PS4. I would like to have my systems save as PS4 because I'm not uploading anything as PC but I can't use NMC for PC in PS4 mode.

So I'm stupid but found out my issue. Do you have any thoughts and/or solutions to what I want to do?

**Comment by jaszhix** [comment]
Nope, its a bug. It will need to be patched.

**Comment by PabloStrange** [comment]
@jaszhix this happened to me to. can you please disable the protection. my username is PabloStrange

**Comment by taukausanake** [comment]
@jaszhix So just to make sure, when I started NMC and protected my username I was using the PC platform option. When I switch to PS4 however it should still use my protected username because computer ID and not any settings, correct? That the bug I take it?

**Comment by jaszhix** [comment]
@PabloStrange 
Unprotected it.
@taukausanake
I checked the server and your username was indeed protected, and I unprotected it. I'm not sure what's going on, but since the feature is clearly buggy, I'd recommend not protecting it and rely on the kindness of random internet people not impersonating your usernames - its kind of a lot of trouble over something with little gain for them anyway. When I added the feature it was supposed to be easier than a username/password, but the next update will probably be using those.
